### PR TITLE
`Plagiarism checks`: Disable plagiarism status update for team exercises

### LINKED ISSUE #8098: `Plagiarism checks`: Plagiarism comparisons of team exercises should not be "Acceptable" or "Rejectable"
### Describe the bug

Artemis does currently not support plagiarism cases for team exercises. It is possible to run the plagiarism detection for team exercises and see the comparisons, though, so an instructor can handle them manually.

However, the buttons and the REST call to accept or reject comparisons, which effectively converts comparisons to cases for non-team exercises, is still available for team exercises at the moment. Clicking them causes an error, as the underlying code is only compatible with non-team exercises.

### To Reproduce

1. Have a team exercise with plagiarism
2. Run the plagiarism detection
3. Confirm a case
4. Artemis returns a 404 because it tries to convert a team name to a user, which fails, and no plagiarism case is created

### Expected behavior

- Artemis should hide the "Accept" and "Reject" buttons in the plagiarism view for team exercises (for now, until full team exercise support is added in the future). 
- Artemis should have an input validation in the corresponding REST call (`PUT /api/courses/{{courseId}}/plagiarism-comparisons/{{comparisonId}}/status`) that rejects the call if the exercise is a team exercise.

### Screenshots

_No response_

### Which version of Artemis are you seeing the problem on?

6.8.3

### What browsers are you seeing the problem on?

Chrome

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I documented the Java code using JavaDoc style.

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.

### Motivation and Context
Fixes #8098 

### Description
This PR disables the Accept and Deny buttons on the client for team exercises. I also added a corresponding check on the server side.

### Steps for Testing
Prerequisites:
- 1 Team Exercise 
- 2 participating teams with similar submissions

1. Run plagiarism checks for the team exercise
2. Verify that when viewing the result of the check, the accept and deny buttons are disabled.

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

## Summary by CodeRabbit

- **New Features**
	- Updated plagiarism check to prevent status updates for team exercises.
- **Bug Fixes**
	- Fixed button disable conditions in the plagiarism header to accurately reflect loading states and team exercise modes.
- **Tests**
	- Added integration tests to verify plagiarism status updates are blocked for team exercises.
- **Refactor**
	- Improved button state management in the plagiarism header by using a more appropriate property (`isLoading`).

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/plagiarism/PlagiarismIntegrationTest.java [hunk 1]
```diff
@@ -119,6 +120,16 @@ void testUpdatePlagiarismComparisonStatus_forbidden_tutor() throws Exception {
...
 
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "editor1", roles = "EDITOR")
+    void testUpdatePlagiarismComparisonStatus_badRequest_teamExercise() throws Exception {
+        textExercise.setMode(ExerciseMode.TEAM);
+        textExercise = textExerciseRepository.save(textExercise);
+
+        request.put("/api/courses/" + course.getId() + "/plagiarism-comparisons/" + plagiarismComparison1.getId() + "/status", new PlagiarismComparisonStatusDTO(NONE),
+                HttpStatus.BAD_REQUEST);
+    }
```
[the remark this answers is not in the document]
Comment (Strohgelaender) [reply src/test/java/de/tum/in/www1/artemis/plagiarism/PlagiarismIntegrationTest.java:131]: The exercise gets loaded on the server. It's the the exercise linked to the plagiarismComparision.

File: src/test/javascript/spec/component/plagiarism/plagiarism-header.component.spec.ts [hunk 2]
```diff
@@ -51,12 +51,12 @@ describe('Plagiarism Header Component', () => {
         comp.confirmPlagiarism();
```
Comment (jakubriegel) [inline src/test/javascript/spec/component/plagiarism/plagiarism-header.component.spec.ts:52]: There could be a test which checks if buttons are disabled for team exercises

**Review by jakubriegel — CHANGES_REQUESTED** [review]

**Review by jakubriegel — COMMENTED** [review]
Tested locally - works fine

Disabled buttons:
image

When I enabled buttons from the console server properly returned an error:
image

**Review by jakubriegel — APPROVED** [review]
All good. Thanks for the update

**Pull request merged**